data_diff/dbt.py

```diff
@@ -0,0 +1,313 @@
+
+        if len(list(diff)) > 0:
+            rich.print(
+                "[red]"
+                + dev_qualified_string
+                + " <> "
+                + prod_qualified_string
+                + "[/] \n"
+                + column_diffs_str
+                + diff.get_stats_string()
+                + "\n"
+            )
+        else:
+            rich.print(
+                "[red]"
+                + dev_qualified_string
+                + " <> "
+                + prod_qualified_string
+                + "[/] \n"
+                + column_diffs_str
+                + "[green]No row differences[/] \n"
+            )
+
+    @staticmethod
+    def _cloud_diff(diff_vars: DiffVars):
+        api_key = os.environ.get("DATAFOLD_API_KEY")
+
+        if diff_vars.datasource_id is None:
+            raise ValueError(
+                "Datasource ID not found, include it as a dbt variable in the dbt_project.yml. \nvars:\n data_diff:\n   datasource_id: 1234"
+            )
+        elif api_key is None:
+            raise ValueError("API key not found, add it as an environment variable called DATAFOLD_API_KEY.")
+        else:
+            url = "https://app.datafold.com/api/v1/datadiffs"
+
+            payload = json.dumps(
+                {
+                    "data_source1_id": diff_vars.datasource_id,
+                    "data_source2_id": diff_vars.datasource_id,
+                    "table1": diff_vars.dev_path,
+                    "table2": diff_vars.prod_path,
+                    "pk_columns": diff_vars.primary_keys,
+                }
+            )
+            headers = {
+                "Authorization": "Key " + api_key,
+                "Content-Type": "application/json",
+            }
+
+            response = requests.request("POST", url, headers=headers, data=payload)
+            response.raise_for_status()
+            data = response.json()
+            id = data["id"]
+            diff_url = f"https://app.datafold.com/datadiffs/{str(id)}/overview"
+            rich.print(
+                "[red]"
+                + ".".join(diff_vars.dev_path)
+                + " <> "
+                + ".".join(diff_vars.prod_path)
+                + "[/] \n    Diff in progress: \n    "
+                + diff_url
+                + "\n"
+            )
+
+
+class DbtParser:
+    DEFAULT_PROFILES_DIR = str(Path.home()) + "/.dbt"
+    DEFAULT_PROJECT_DIR = str(os.getcwd())
+
+    def __init__(self, profiles_dir_override: str, project_dir_override: str, is_cloud: bool):
+        self.profiles_dir = profiles_dir_override or self.DEFAULT_PROFILES_DIR
+        self.project_dir = project_dir_override or self.DEFAULT_PROJECT_DIR
+        self.is_cloud = is_cloud
+        self.connection = None
+        self.project_dict = None
+        self.requires_upper = False
+
+    def get_datadiff_variables(self):
+        try:
+            vars = self.project_dict.get("vars").get("data_diff")
+            assert vars is not None
```

Comment:
There can be a case when `vars:` is defined, but it is not a dict/object. It will fail with an exception anyway. So, It might be a good idea to always fail with an explanatory `KeyError` instead of a cryptic `AssertionError`.

```suggestion
            vars = self.project_dict["vars"]["data_diff"]
```

Reply:
fixed